libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1613,6 +1613,13 @@ bool TypeChecker::visit(UnaryOperation const& _operation)
 	_operation.annotation().isPure = !modifying && *_operation.subExpression().annotation().isPure;
 	_operation.annotation().isLValue = false;
 
+	if (subExprType->nameable() && subExprType->containsNestedMapping())
```

Comment:
What about something like
```solidity
contract C {
    mapping (uint => uint) m;
    function f() external {
        m[1]++;
    }
}
```
Wouldn't that also make it here?

Reply:
uh. You're right I forgot to actually check for the delete operation :sweat_smile:

Reply:
Actually no. Your example works fine because `m[1]` does NOT contain a nested mapping. It would need to be
```
contract C {
    mapping (uint => uint) m;
    function f() external {
        m++;
    }
}
```
which isn't supported..

Reply:
However, I should still add a check for the `Delete` token